web-api/src/mind/state.rs

```diff
@@ -69,9 +71,9 @@ impl State {
     pub(super) async fn insert(&self, documents: Vec<Document>) -> Result<(), Panic> {
         let documents = documents
             .into_iter()
-            .map(|document| {
-                let embedding = self.embedder.run(&document.snippet)?;
-                Ok(IngestedDocument {
+            .map(|document| async move {
+                let embedding = self.embedder.run(&document.snippet).await?;
+                Ok::<IngestedDocument, Panic>(IngestedDocument {
```

Comment:
:art: it should probably be enough to hint at the error type
```suggestion
                Ok::<_, Panic>(IngestedDocument {
```